src/rdb.c

```diff
@@ -2719,8 +2734,8 @@ void backgroundSaveDoneHandler(int exitcode, int bysignal) {
  * the child did not exit for an error, but because we wanted), and performs
  * the cleanup needed. */
 void killRDBChild(void) {
-    kill(server.rdb_child_pid,SIGUSR1);
-    rdbRemoveTempFile(server.rdb_child_pid, 0);
+    kill(server.child_pid, SIGUSR1);
```

Comment:
how come killRDBChild doesn't call resetChildState like the killer of AOF and modules?

Reply:
ok, i realized that unlike killAppendOnlyChild and TerminateModuleForkChild, the killRDBChild doesn't call `wait4`.
so the fact it doesn't clear the pid variable, is desired, and that later on checkChildrenDone will find it and do proper cleanup.

looking at redis 6.0.9, i see that same as here, it does do rdbRemoveTempFile, closeChildInfoPipe, and updateDictResizePolicy.
all of which are unnecessary, but also don't cause any harm.
closeChildInfoPipe, and updateDictResizePolicy will by done by checkChildrenDone.
and rdbRemoveTempFile will be done by backgroundSaveDoneHandlerDisk (only in case of `bysignal`) which is the case discussed here (killRDBChild).

all of these 3 do no harm when called excessively (so there's no bug in 6.0.9).
- closeChildInfoPipe has built in protection against being called twice.
- updateDictResizePolicy has no effect.
- rdbRemoveTempFile will call bg_unlink, and ignore the error it returns.

so my advise is:
1. no bug in 6.0, let's leave it as is.
2. in this PR, we'll remove these 3 lines from killAppendOnlyChild and add a big comment explaining that they'll be done by checkChildrenDone later on.

Reply:
Hi @oranagra `killRDBChild` also be called when redis shutdown. so we should call `rdbRemoveTempFile` when `killRDBChild`. Tcl regression tests failed in shutdown.tcl.
If call `rdbRemoveTempFile` in `killRDBChild`, as you said, we may call `rdbRemoveTempFile` twice, but it does no obvious harm. (`opne` and `link` fail in `bg_unlink`). Of course, we can call extra `rdbRemoveTempFile` when redis shutdown, but i prefer to keep `rdbRemoveTempFile` in `killRDBChild`

Reply:
@ShooterIT considering there's already a big comment in prepareForShutdown talking about a call to rdbRemoveTempFile and how it behaves, i think it should be there now.